src/server.c

```diff
@@ -5167,6 +5173,31 @@ sds genRedisInfoStringCommandStats(sds info, dict *commands) {
     return info;
 }
 
+/* Writes the ACL metrics to the info */
+sds genRedisInfoStringACLStats(sds info) {
+    sds metrics = sdsempty();
+    
+    metrics = sdscatprintf(metrics,
+    "auth_failure_times:%ld\r\n",
+    server.acl_info.user_auth_failures);
+    
+    metrics = sdscatprintf(metrics,
+    "invalid_cmd_accesses:%ld\r\n",
+    server.acl_info.invalid_cmd_accesses);
+    
+    metrics = sdscatprintf(metrics,
+    "invalid_key_accesses:%ld\r\n",
+    server.acl_info.invalid_key_accesses);
+
+    metrics = sdscatprintf(metrics,
+    "invalid_channel_accesses:%ld\r\n",
+    server.acl_info.invalid_channel_accesses);
+
+    info = sdscatprintf(info, "%s", metrics);
```

Comment:
i'd rather this be one big catprintf (like the ones in server.c) without a temporary string.

also, i think i'm not happy with the info field names, i think they should have some common prefix.
considering that we might add more of these in the future, maybe the common prefix should be long and the suffix short (like command stats and error stats), so it'll be easy to understand these are counters of different errors coming from the same mechanism.

Reply:
I was going to mention the naming last, since that is the easiest thing to change. I was thinking of something like:
```
acl_invalid_auth
acl_invalid_cmd_permission
acl_invalid_key_permission
acl_invalid_channel_permission
```

Something like `acl_invalid_permission_auth/cmd/key/channel` would technically have a longer prefix, but it seems less clear.

Reply:
how about:
`acl_deny_access_[auth,cmd,key,chan]`?
p.s. technically, an AUTH failure is not necessarily an ACL (could be legacy password), but i think we can ignore that.

Reply:
The only objection I have there is the info is almost all past tense, so:
`acl_access_denied_[auth/cmd/key/chan]`